File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/ElseClauseSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class ElseClauseSyntax(val elseKeyword: SyntaxToken, val elseStatement: StatementSyntax) : SyntaxNode() {
    override val kind: SyntaxKind
        get() = SyntaxKind.ElseClause
    override val children: List<SyntaxNode>
        get() = listOf(elseKeyword, elseStatement)
}

File: cs/Minsk/CodeAnalysis/Syntax/ElseClauseSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public class ElseClauseSyntax : SyntaxNode
{
    public ElseClauseSyntax(SyntaxToken elseKeyword, StatementSyntax elseStatement)
    {
        ElseKeyword = elseKeyword;
        ElseStatement = elseStatement;
    }

    public override SyntaxKind Kind => SyntaxKind.ElseClause;

    public override IEnumerable<SyntaxNode> Children => new SyntaxNode[] { ElseKeyword, ElseStatement };

    public SyntaxToken ElseKeyword { get; }
    public StatementSyntax ElseStatement { get; }
}
